spark-extension-shims-spark3/src/test/scala/org/apache/spark/sql/auron/AuronCheckConvertBroadcastExchangeSuite.scala

```diff
@@ -0,0 +1,177 @@
+import org.apache.spark.sql.execution.exchange.BroadcastExchangeExec
+import org.apache.spark.sql.test.SharedSparkSession
+
+class AuronCheckConvertBroadcastExchangeSuite
+    extends QueryTest
+    with SharedSparkSession
+    with AuronSQLTestHelper
+    with org.apache.spark.sql.execution.adaptive.AdaptiveSparkPlanHelper {
+
+  test(
+    "test bhj broadcastExchange to native where spark.auron.enable.broadcastexchange is true") {
+    val spark = SparkSession
+      .builder()
+      .master("local[2]")
+      .appName("checkConvertToNativeShuffleManger")
+      .config("spark.sql.shuffle.partitions", "4")
+      .config("spark.sql.autoBroadcastJoinThreshold", -1)
+      .config("spark.sql.extensions", "org.apache.spark.sql.auron.AuronSparkSessionExtension")
+      .config(
+        "spark.shuffle.manager",
+        "org.apache.spark.sql.execution.auron.shuffle.AuronShuffleManager")
+      .config("spark.memory.offHeap.enabled", "false")
+      .config("spark.auron.enable", "true")
+      .getOrCreate()
+
+    spark.sql(
+      "create table if not exists broad_cast_table1 using parquet PARTITIONED BY (part) as select 1 as c1, 2 as c2, 'test test' as part")
+    spark.sql(
+      "create table if not exists broad_cast_table2 using parquet PARTITIONED BY (part) as select 1 as c1, 2 as c2, 'test test' as part")
+    val executePlan =
+      spark.sql(
+        "select /*+ broadcast(a)*/ a.c1, a.c2 from broad_cast_table1 a inner join broad_cast_table2 b on a.c1 = b.c1")
+
+    val plan = executePlan.queryExecution.executedPlan.asInstanceOf[AdaptiveSparkPlanExec]
+    val broadcastExchangeExec =
+      plan.executedPlan
+        .collectFirst { case broadcastExchangeExec: BroadcastExchangeExec =>
+          broadcastExchangeExec
+        }
+
+    val afterConvertPlan = AuronConverters.convertSparkPlan(broadcastExchangeExec.get)
+    assert(afterConvertPlan.isInstanceOf[NativeBroadcastExchangeExec])
+    checkAnswer(executePlan, Seq(Row(1, 2)))
+  }
+
+  test(
+    "test bnlj broadcastExchange to native where spark.auron.enable.broadcastexchange is true") {
+    val spark = SparkSession
+      .builder()
+      .master("local[2]")
+      .appName("checkConvertToNativeShuffleManger")
+      .config("spark.sql.shuffle.partitions", "4")
+      .config("spark.sql.autoBroadcastJoinThreshold", -1)
+      .config("spark.sql.extensions", "org.apache.spark.sql.auron.AuronSparkSessionExtension")
+      .config(
+        "spark.shuffle.manager",
+        "org.apache.spark.sql.execution.auron.shuffle.AuronShuffleManager")
+      .config("spark.memory.offHeap.enabled", "false")
+      .config("spark.auron.enable", "true")
+      .getOrCreate()
+
+    spark.sql(
+      "create table if not exists broad_cast_table1 using parquet PARTITIONED BY (part) as select 1 as c1, 2 as c2, 'test test' as part")
+    spark.sql(
+      "create table if not exists broad_cast_table2 using parquet PARTITIONED BY (part) as select 1 as c1, 2 as c2, 'test test' as part")
```

Comment:
suggestion: should we use temp views here instead? I think it’s safer and faster to build the data in memory and register it as a temp view using createOrReplaceTempView, as temp views are session-scoped. 

```scala
private def createData()(implicit spark: SparkSession): Unit = {
  import spark.implicits._
  Seq((1, 2, "test test")).toDF("c1","c2","part").createOrReplaceTempView("t1")
  Seq((1, 2, "test test")).toDF("c1","c2","part").createOrReplaceTempView("t2")
}
```